plugins/obs-ffmpeg/obs-ffmpeg-mux.c

```diff
@@ -465,10 +358,31 @@ static int deactivate(struct ffmpeg_muxer *stream, int code)
 	}
 
 	os_atomic_set_bool(&stream->stopping, false);
+
+	if (stream->is_hls) {
+		if (stream->mux_thread_joinable) {
+			os_event_signal(stream->stop_event);
+			os_sem_post(stream->write_sem);
+			pthread_join(stream->mux_thread, NULL);
+			stream->mux_thread_joinable = false;
+		}
+
+		pthread_mutex_lock(&stream->write_mutex);
+
+		while (stream->packets.size) {
+			struct encoder_packet packet;
+			circlebuf_pop_front(&stream->packets, &packet,
+					    sizeof(packet));
+			obs_encoder_packet_release(&packet);
+		}
+
+		pthread_mutex_unlock(&stream->write_mutex);
+	}
+
 	return ret;
 }
 
-static void ffmpeg_mux_stop(void *data, uint64_t ts)
+void ffmpeg_mux_stop(void *data, uint64_t ts)
```

Comment:
What if ts != 0? We may need a custom stop function for hls.

Reply:
I'll look into this